go/controller/internal/httpserver/handlers/invoke.go

```diff
@@ -0,0 +1,166 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/mux"
+	autogen_client "github.com/kagent-dev/kagent/go/autogen/client"
+	"github.com/kagent-dev/kagent/go/controller/internal/httpserver/errors"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// mockAutogenClient interface for testing
+type mockAutogenClient interface {
+	CreateSession(*autogen_client.CreateSession) (*autogen_client.Session, error)
+	CreateRun(*autogen_client.CreateRunRequest) (*autogen_client.CreateRunResult, error)
+}
+
+// InvokeHandler handles agent invocation requests
+type InvokeHandler struct {
+	*Base
+	testClient interface{}
+}
+
+// NewInvokeHandler creates a new InvokeHandler
+func NewInvokeHandler(base *Base) *InvokeHandler {
+	return &InvokeHandler{Base: base}
+}
+
+// SetTestClient allows setting a mock client for testing
+func (h *InvokeHandler) SetTestClient(client interface{}) {
+	h.testClient = client
+}
+
+// InvokeRequest represents a request to invoke an agent
+type InvokeRequest struct {
+	Message string                 `json:"message"`
+	Sync    bool                   `json:"sync"`
+	Context map[string]interface{} `json:"context,omitempty"`
+	UserID  string                 `json:"user_id,omitempty"`
+}
+
+// InvokeResponse represents a response from an agent invocation
+type InvokeResponse struct {
+	SessionID   string `json:"sessionId"`
+	Response    string `json:"response,omitempty"`
+	StatusURL   string `json:"statusUrl,omitempty"`
+	Status      string `json:"status"`
+	CompletedAt string `json:"completedAt,omitempty"`
+}
+
+// HandleInvokeAgent handles POST /api/agents/{agentId}/invoke requests
+func (h *InvokeHandler) HandleInvokeAgent(w ErrorResponseWriter, r *http.Request) {
+	log := ctrllog.FromContext(r.Context()).WithName("invoke-handler").WithValues("operation", "invoke")
+
+	// Extract agent ID from path
+	vars := mux.Vars(r)
+	agentIDStr := vars["agentId"]
+	if agentIDStr == "" {
+		w.RespondWithError(errors.NewBadRequestError("Agent ID is required", nil))
+		return
+	}
+	
+	// Convert agent ID to int
+	agentID, err := strconv.Atoi(agentIDStr)
+	if err != nil {
+		if httpWriter, ok := w.(http.ResponseWriter); ok {
+			httpWriter.WriteHeader(http.StatusBadRequest)
+		}
+		w.RespondWithError(errors.NewBadRequestError("Invalid agent ID format, must be an integer", err))
+		return
+	}
+	log = log.WithValues("agentId", agentID)
+
+	// Parse request body
+	var invokeRequest InvokeRequest
+	if err = DecodeJSONBody(r, &invokeRequest); err != nil {
+		w.RespondWithError(errors.NewBadRequestError("Invalid request body", err))
+		return
+	}
+
+	// Get user ID from request or header
+	userID := invokeRequest.UserID
```

Comment:
check the GetUserID here https://github.com/kagent-dev/kagent/blob/main/go/controller/internal/httpserver/handlers/helpers.go